source/particle/world.cc

```diff
@@ -318,10 +336,11 @@ namespace aspect
                              const void *data)
     {
       const unsigned int *n_particles_in_cell = static_cast<const unsigned int *> (data);
```

Comment:
This variable is poorly named because it doesn't denote a number but a *pointer* into an array of numbers. Maybe `particles_in_cell_ptr` or similar?